qCC/mainwindow.h

```diff
@@ -59,592 +59,2410 @@ class ccTracePolylineTool;
 
 struct dbTreeSelectionInfo;
 
-namespace Ui {
+namespace Ui
+{
 	class MainWindow;
-} 
+}
 
 //! Main window
-class MainWindow : public QMainWindow, public ccMainAppInterface, public ccPickingListener 
+/**
+ * \class MainWindow
+ * \brief The main window class for the application.
+ *
+ * This class represents the main window of the application. It inherits from QMainWindow and implements various interfaces such as ccMainAppInterface and ccPickingListener. The main window contains the user interface elements and handles user interactions.
+ */
+class MainWindow : public QMainWindow, public ccMainAppInterface, public ccPickingListener
 {
+	/**
+	 * \brief The Q_OBJECT macro is used in the declaration of a class that implements signals and slots.
+	 *
+	 * The Q_OBJECT macro must appear in the private section of a class definition that declares its own signals and slots or that uses other services provided by Qt's meta-object system.
+	 * It is necessary for the signals and slots mechanism to work properly.
+	 */
 	Q_OBJECT
 
 protected:
-	//! Default constructor
+	/**
+	 * \brief Constructor for the MainWindow class.
+	 *
+	 * Initializes the MainWindow object and sets up the user interface.
+	 * Creates various member variables and connects signals and slots.
+	 * Sets the window title, populates menus, and initializes console.
+	 * Creates advanced widgets and sets up dynamic menus.
+	 * Creates ccDBRoot object for the database tree view.
+	 * Creates QMdiArea object for the MDI area.
+	 * Creates ccPickingHub object for picking functionality.
+	 * Restores the state of the 'auto-restore' menu entry.
+	 * Connects actions, creates a new 3D view, sets up input devices.
+	 * Unfreezes the user interface, updates the UI, and displays a status message.
+	 * Prints a message to the console indicating that CloudCompare has started.
+	 */
 	MainWindow();
 
-	//! Default desctructor
+	/**
+	 * \brief Destructor for the MainWindow class.
+	 *
+	 * This destructor is responsible for cleaning up any resources used by the MainWindow object.
+	 * It overrides the base class destructor.
+	 */
 	~MainWindow() override;
-	
+
 public:
-	//! Returns the unique instance of this object
-	static MainWindow* TheInstance();
+	/**
+	 * \brief Returns the instance of the MainWindow class.
+	 *
+	 * \return A pointer to the MainWindow instance.
+	 */
+	static MainWindow *TheInstance();
```

Comment:
normally in CloudCompare we add the '*' just after the class/typename (Maindow *TheInstance() --> Maindow* TheInstance()).

Is it possible to fix this behavior?

Reply:
yes I'll change the way we do the pointers

Reply:
I'm having some difficulties with my code styler I'll get this done as soon as I can,